src/odemis/gui/cont/tabs.py

```diff
@@ -316,6 +316,516 @@ def get_display_priority(cls, main_data):
+            tab_data.spotStream = spot_stream
+            # TODO: add to tab_data.streams and move the handling to the stream controller?
+            tab_data.spotPosition.subscribe(self._onSpotPosition)
+            tab_data.tool.subscribe(self.on_tool_change)
+
+        self._settingbar_controller = settings.LocalizationSettingsController(
+            panel,
+            tab_data
+        )
+
+        self._streambar_controller = streamcont.SecomStreamsController(
+            tab_data,
+            panel.pnl_secom_streams,
+            view_ctrl=self.view_controller
+        )
+
+        self._acquisition_controller = acqcont.OverviewStreamAcquiController(tab_data, panel)
+
+        self._overview_stream_controller = streamcont.StreamBarController(
+            tab_data,
+            panel.pnl_overview_streams,
+            view_ctrl=self.view_controller,
+        )
+
+        self._overview_stream_controller.add_action("Acqurie...", self._on_acquire)
+
+        # Toolbar
+        self.tb = panel.secom_toolbar
+        for t in TOOL_ORDER:
+            if t in tab_data.tool.choices:
+                self.tb.add_tool(t, tab_data.tool)
+        # Add fit view to content to toolbar
+        self.tb.add_tool(TOOL_ACT_ZOOM_FIT, self.view_controller.fitViewToContent)
+        # auto focus
+        self._autofocus_f = model.InstantaneousFuture()
+        self.tb.add_tool(TOOL_AUTO_FOCUS, self.tab_data_model.autofocus_active)
+        self.tb.enable_button(TOOL_AUTO_FOCUS, False)
+        self.tab_data_model.autofocus_active.subscribe(self._onAutofocus)
+        tab_data.streams.subscribe(self._on_current_stream)
+
+        # To automatically play/pause a stream when turning on/off a microscope,
+        # and add the stream on the first time.
+        if hasattr(tab_data, 'opticalState'):
+            tab_data.opticalState.subscribe(self.onOpticalState)
+            if main_data.ccd:
+                self._add_opt_stream = self._streambar_controller.addFluo
+            elif main_data.photo_ds:
+                # Use the first photo-detector in alphabetical order
+                pd0 = min(main_data.photo_ds, key=lambda d: d.role)
+                self._add_opt_stream = partial(self._streambar_controller.addConfocal,
+                                               detector=pd0)
+            else:
+                logging.error("No optical detector found")
+
+        if hasattr(tab_data, 'emState'):
+            tab_data.emState.subscribe(self.onEMState)
+            # decide which kind of EM stream to add by default
+            if main_data.sed:
+                self._add_em_stream = self._streambar_controller.addSEMSED
+            elif main_data.bsd:
+                self._add_em_stream = self._streambar_controller.addSEMBSD
+            else:
+                logging.error("No EM detector found")
+
+        # Create streams before state controller, so based on the chamber state,
+        # the streams will be enabled or not.
```

Comment:
You've removed the creation of the state controller. That's fine, but then you can also remove this comment, and more importantly the code related to the "opticalState" and "emState". You can also remove the LOAD, SEM and OPTICAL buttons.